### PR TITLE
Angus/basic fits frame tests

### PR DESCRIPTION
**Pull request by veggiesaurus**
#797 will result in a large number of new unit tests, many of which will be using the `Frame` class. Here's an initial example of some unit tests for loading various simple test FITS images (2D, 3D and 4D).

I've also added an example of a "friend" test for accessing protected data within `Frame`.

### CODE REVIEW

File: test/TestFitsImage.cc [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+TEST_F(FitsImageTest, ExampleFriendTest) {
+    auto path_string = ImagePath("noise_10px_10px.fits");
+    // TestFrame used instead of Frame if access to protected values is required
+    std::unique_ptr<TestFrame> frame(new TestFrame(0, carta::FileLoader::GetLoader(path_string), "0"));
+    EXPECT_TRUE(frame->IsValid());
+    EXPECT_TRUE(frame->_open_image_error.empty());
+}
+
+TEST_F(FitsImageTest, CorrentShape2dImage) {
```
Comment (confluence) [inline test/TestFitsImage.cc]: "Corrent" typo

File: test/TestFitsImage.cc [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+    auto shape = frame->ImageShape();
+    EXPECT_EQ(shape.size(), 2);
+    EXPECT_EQ(shape[0], 10);
+    EXPECT_EQ(shape[1], 10);
+    EXPECT_EQ(frame->Depth(), 1);
+    EXPECT_EQ(frame->NumStokes(), 1);
+}
+
+TEST_F(FitsImageTest, CorrentShape3dImage) {
```
Comment (confluence) [inline test/TestFitsImage.cc]: "Corrent" typo

File: test/TestFitsImage.cc [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    EXPECT_EQ(shape.size(), 3);
+    EXPECT_EQ(shape[0], 10);
+    EXPECT_EQ(shape[1], 10);
+    EXPECT_EQ(shape[2], 10);
+    EXPECT_EQ(frame->Depth(), 10);
+    EXPECT_EQ(frame->NumStokes(), 1);
+    EXPECT_EQ(frame->StokesAxis(), -1);
+}
+
+TEST_F(FitsImageTest, CorrentShapeDegenerate3dImages) {
```
Comment (confluence) [inline test/TestFitsImage.cc]: "Corrent" typo

File: test/TestFitsImage.cc [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    EXPECT_EQ(shape[0], 10);
+    EXPECT_EQ(shape[1], 10);
+    EXPECT_EQ(shape[2], 1);
+    EXPECT_EQ(shape[3], 10);
+    EXPECT_EQ(frame->Depth(), 10);
+    EXPECT_EQ(frame->NumStokes(), 1);
+    EXPECT_EQ(frame->StokesAxis(), 2);
+}
+
+TEST_F(FitsImageTest, CorrentShape4dImages) {
```
Comment (confluence) [inline test/TestFitsImage.cc]: "Corrent" typo

**Review by confluence — CHANGES_REQUESTED** [review]
This looks good so far; I just found some typos.

File: test/TestFitsImage.cc (hunk 4 above)
Comment (veggiesaurus) [reply test/TestFitsImage.cc]: :laughing: oops. At least I was consistent!

**Review by confluence — APPROVED** [review]

**Pull request merged**